packages/core/src/types.ts

```diff
@@ -193,3 +193,8 @@ export type FlatArray<Arr> = Arr extends ReadonlyArray<infer InnerArr> ? FlatArr
  * Get the awaited (resolved promise) type of Promise type.
  */
 export type Awaited<T> = T extends Promise<infer U> ? U : never
+
+/**
+ * Type util that returns `true` or `false` based on whether the input type `T` is of type `any`
+ */
+export type IsAny<T> = unknown extends T ? ([keyof T] extends [never] ? false : true) : false
```

Comment:
Looking at this makes me think that this should not work, but it does haha. Quite interesting.

Reply:
Yeah this comes straight from Stack overflow. No idea why it works. I can add a link to th so post though